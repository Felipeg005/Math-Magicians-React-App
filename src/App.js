import './App.css';
import Calculator from './components/calculator';

function App() {
  const newCalculator = new Calculator();
  return (
    newCalculator.render()
  );
}

export default App;
