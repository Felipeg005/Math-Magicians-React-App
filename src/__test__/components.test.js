import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Calculator from '../components/calculator';
import Home from '../pages/home';
import Quote from '../pages/quote';
import App from '../App';

describe('Test snapshot', () => {
  test('Render Home', () => {
    const component = render(<Home />);
    expect(component.container).toHaveTextContent('Welcome to our page!');
  });

  test('Render Calculator', () => {
    const component = render(<Calculator />);
    expect(component.container).toHaveTextContent('Let`s do some Math!');
    expect(component.container).toHaveTextContent('1');
    expect(component.container).toHaveTextContent('0');
    expect(component.container).toHaveTextContent('AC');
  });

  test('Render Calculator', () => {
    const component = render(<App />);
    expect(component.container).toHaveTextContent('Home');
    expect(component.container).toHaveTextContent('Calculator');
    expect(component.container).toHaveTextContent('Quote');
  });

  test('Render Quote', () => {
    const component = render(<Quote />);
    expect(component.container).toHaveTextContent('Albert Einstein');
  });

  test('Snapshot', () => {
    const component = renderer.create(< App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
