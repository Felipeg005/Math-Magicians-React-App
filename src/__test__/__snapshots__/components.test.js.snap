// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Snapshot 1`] = `
<nav
  className="nav-container"
>
  <h1
    className="logo"
  >
    Math Magicians
  </h1>
  <ul
    className="menu-container"
  >
    <a
      href="/Math-Magicians-React-App/"
      onClick={[Function]}
      style={
        Object {
          "color": "white",
        }
      }
    >
      <li>
        Home
      </li>
    </a>
    <a
      href="/Math-Magicians-React-App/calculator"
      onClick={[Function]}
      style={
        Object {
          "color": "white",
        }
      }
    >
      <li>
        Calculator
      </li>
    </a>
    <a
      href="/Math-Magicians-React-App/quote"
      onClick={[Function]}
      style={
        Object {
          "color": "white",
        }
      }
    >
      <li>
        Quote
      </li>
    </a>
  </ul>
</nav>
`;
